providers/in-process-flagd/provider.go

```diff
@@ -0,0 +1,644 @@
+	logger                *logger.Logger
+	evaluator             eval.IEvaluator
+	syncSource            sync.ISync
+	mu                    sync2.Mutex
+	ofEventChannel        chan of.Event
+}
+type ProviderConfiguration struct {
+	CertificatePath string
+	TLSEnabled      bool
+	SourceConfig    runtime.SourceConfig
+}
+
+type ProviderOption func(*Provider)
+
+func NewProvider(ctx context.Context, opts ...ProviderOption) *Provider {
+	ctx, cancel := context.WithCancel(ctx)
+	provider := &Provider{
+		ctx:        ctx,
+		cancelFunc: cancel,
+		// providerConfiguration maintains its default values, to ensure that the FromEnv option does not overwrite any explicitly set
+		// values (default values are then set after the options are run via applyDefaults())
+		providerConfiguration: &ProviderConfiguration{},
+		isReady:               make(chan struct{}),
+		connectionInfo: connectionInfo{
+			state:           stateNotReady,
+			retries:         0,
+			maxSyncRetries:  defaultMaxSyncRetries,
+			backoffDuration: defaultBackoffDuration,
+		},
+		ofEventChannel: make(chan of.Event),
+		logger:         logger.NewLogger(nil, false),
+	}
+	provider.applyDefaults()   // defaults have the lowest priority
+	FromEnv()(provider)        // env variables have higher priority than defaults
+	for _, opt := range opts { // explicitly declared options have the highest priority
+		opt(provider)
+	}
+
+	if provider.providerConfiguration.SourceConfig.URI == "" {
+		log.Fatal(errors.New("no sync source configuration provided"))
+	}
+
+	s := store.NewFlags()
+
+	s.FlagSources = append(s.FlagSources, provider.providerConfiguration.SourceConfig.URI)
+	s.SourceMetadata[provider.providerConfiguration.SourceConfig.URI] = store.SourceDetails{
+		Source:   provider.providerConfiguration.SourceConfig.URI,
+		Selector: provider.providerConfiguration.SourceConfig.Selector,
+	}
+
+	var err error
+	provider.syncSource, err = syncProviderFromConfig(provider.logger, provider.providerConfiguration.SourceConfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	provider.evaluator = eval.NewJSONEvaluator(
+		provider.logger,
+		s,
+		eval.WithEvaluator(
+			"fractionalEvaluation",
+			eval.NewFractionalEvaluator(provider.logger).FractionalEvaluation,
+		),
+		eval.WithEvaluator(
+			"starts_with",
+			eval.NewStringComparisonEvaluator(provider.logger).StartsWithEvaluation,
+		),
+		eval.WithEvaluator(
+			"ends_with",
+			eval.NewStringComparisonEvaluator(provider.logger).EndsWithEvaluation,
+		),
+		eval.WithEvaluator(
+			"sem_ver",
+			eval.NewSemVerComparisonEvaluator(provider.logger).SemVerEvaluation,
+		),
+	)
+
+	dataSync := make(chan sync.DataSync, 1)
+
+	go provider.watchForUpdates(dataSync)
+
+	if err := provider.syncSource.Init(provider.ctx); err != nil {
+		log.Fatal("sync provider Init returned error: %w", err)
+	}
+
+	// Start sync provider
+	go provider.startSyncSources(dataSync)
+
+	return provider
+}
+
+func (p *Provider) startSyncSources(dataSync chan sync.DataSync) {
```

Comment:
```suggestion
func (p *Provider) startSyncSource(dataSync chan sync.DataSync) {
```

nit: seems like this is just one source, so perhaps this is a better name?